rollup.config.js

```diff
@@ -95,4 +95,11 @@ export default [
   createCommonJSConfig('src/shallow.ts', 'dist/shallow.js'),
   createCommonJSConfig('src/middleware.ts', 'dist/middleware.js'),
   createCommonJSConfig('src/vanilla.ts', 'dist/vanilla.js'),
+
+  // Create a purely ESM version, can be imported from `zustand/esm`
+  createESMConfig('src/vanilla.ts', 'dist/esm/vanilla.js'),
+  createESMConfig('src/shallow.ts', 'dist/esm/shallow.js'),
+  createESMConfig('src/index.ts', 'dist/esm/index.js'),
```

Comment:
```suggestion
  createESMConfig('src/vanilla.ts', 'dist/vanilla.module.js'),
  createESMConfig('src/shallow.ts', 'dist/shallow.module.js'),
  createESMConfig('src/index.ts', 'dist/index.module.js'),
```
Does this make sense? I'm not sure if "module" and "esm" is the same thing...
Anyway, my point is we want to make it consistent among zustand/jotai/valtio.
(I mean we could change the other two, if that's better.)

Reply:
looks good too. I have no strong opinion. I assume you have a broader overview and insights from users for what suits best.

Reply:
As I understand, usually "module" and "esm" refer the same thing, but "mjs" does not.

Reply:
same here. mjs is just the workaround of nodejs that exist(ed) while nodejs was figuring out how to move to real ESM